### PR TITLE
Mini map

### PR DESCRIPTION
**Pull request by krahab**
Added the minimap and many decorative blocks

### CODE REVIEW

File: src/main/java/net/jesteur/me/client/screens/MiddleEarthMapScreen.java [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+        float transformedPosY = ((posZ / WORLD_SIZE.y) * MAP_WINDOW_HEIGHT * getZoomLevel()) - mapDisplacementY;
+
+        transformedPosX = Math.max(textureOffsetX, Math.min(MAP_WINDOW_WIDTH - textureOffsetX, transformedPosX));
+        transformedPosY = Math.max(textureOffsetY, Math.min(MAP_WINDOW_HEIGHT - textureOffsetY, transformedPosY));
+
+        return new Vec2f(transformedPosX, transformedPosY);
+    }
+
+    private boolean cursorIsOutsideOfMapBounds(double mouseX, double mouseY){
+        return !(mouseX >= 275 && mouseX <= 685 && mouseY >= 75 && mouseY <= 430);
```
Comment (Jukoz) [inline src/main/java/net/jesteur/me/client/screens/MiddleEarthMapScreen.java:304]: Not good, we need to use `(this.width / 2) - (MAP_WINDOW_WIDTH / 2)`

File: src/main/java/net/jesteur/me/entity/spider/MirkwoodSpiderEntity.java [hunk 2]
```diff
@@ -142,7 +144,6 @@ public void slowMovement(BlockState state, Vec3d multiplier) {
         if (!state.isOf(Blocks.COBWEB)) {
```
Comment (Jukoz) [inline src/main/java/net/jesteur/me/entity/spider/MirkwoodSpiderEntity.java:144]: We might need to add other cobweb blocks

File: src/main/java/net/jesteur/me/client/screens/MiddleEarthMapScreen.java (hunk 1 above)
Comment (krahab) [reply src/main/java/net/jesteur/me/client/screens/MiddleEarthMapScreen.java]: Fixed

File: src/main/java/net/jesteur/me/entity/spider/MirkwoodSpiderEntity.java (hunk 2 above)
Comment (krahab) [reply src/main/java/net/jesteur/me/entity/spider/MirkwoodSpiderEntity.java]: Fixed

**Review by Jukoz — APPROVED** [review]
Great work!

**Pull request merged**